### DISCUSSION TITLE
Undocumented TaskGroup feature

### DISCUSSION THREAD

**Discussion by shay1bz** [apache/airflow#29291] [Q&A]
### What do you see as an issue?

[Original SO question](https://stackoverflow.com/questions/75234917/using-airflow-taskgroup-without-setting-direct-dependencies)
TaskGroup allows to set direct dependencies on the DAG's "outer" tasks:
```
with DAG(...) as dag:
    t1 = DummyOperator(task_id="t1")
    t2 = DummyOperator(task_id="t2")
    t3 = DummyOperator(task_id="t3")
    t4 = DummyOperator(task_id="t4")

    with TaskGroup(group_id="myTG"):
        tg1 = DummyOperator(task_id="TG1")
        tg2 = DummyOperator(task_id="TG2")
        tg3 = DummyOperator(task_id="TG3")

        ##########################################################
        # setting direct dependencies on the "outer" DAG tasks: #
        ##########################################################
        tg1.set_upstream(t2)    
        tg2.set_upstream(t4)  

        # internal TG structure: 
        tg1 >> tg3
        tg2 >> tg3

    t1 >> t2
    t3 >> t4
```

And the outcome is as expected:
![Screenshot from 2023-01-25 15-32-56](https://user-images.githubusercontent.com/35406867/214586247-62036e1d-b556-4abd-972f-a7c5c2a1b421.png)

But this behavior is not documented and users might get the impression that this is some undefined behavior. TMHO this is a very nice feature which should be explicitly documented

### Solving the problem

Adding a small paragraph in the TG doc section, describing that TG supports such tasks wiring, and not just the common configuration.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by Taragolis** [comment]
Feel free to make a PR, and make documentation more clear about this behaviour
